### PR TITLE
bpo-35952: Fix pythoninfo when the compiler is missing

### PR DESCRIPTION
**Pull request by xdegaye**
https://bugs.python.org/issue35952

### CODE REVIEW

File: Lib/test/pythoninfo.py [hunk 1]
```diff
@@ -571,13 +571,23 @@ def collect_cc(info_add):
...
+        # Cannot run the compiler, for example when Python has been
+        # cross-compiled and installed on the target platform where the
+        # compiler is missing.
+        pass
+    else:
+        stdout = proc.communicate()[0]
+
+    # Failed to run the compiler or 'CC --version' failed.
+    if proc is None or proc.returncode:
+        info_add('CC.name', normalize_text(CC))
```
Comment (vstinner) [inline Lib/test/pythoninfo.py]: I'm not sure of the purpose of this change: CC is already logged as "sysconfig[CC]: gcc -pthread", no?

File: Lib/test/pythoninfo.py [hunk 2]
```diff
@@ -571,13 +571,23 @@ def collect_cc(info_add):
...
-    stdout = proc.communicate()[0]
-    if proc.returncode:
-        # CC --version failed: ignore error
+    proc = None
+    try:
+        proc = subprocess.Popen(args,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.STDOUT,
+                                universal_newlines=True)
+    except (FileNotFoundError, PermissionError):
```
Comment (vstinner) [inline Lib/test/pythoninfo.py]: This part is perfectly fine ;-)

Comment (vstinner) [reply Lib/test/pythoninfo.py]: Hum, honestly, ignoring any OSError would be just fine. There is no need to be specific here.

File: Lib/test/pythoninfo.py (hunk 1 above)
Comment (xdegaye) [reply Lib/test/pythoninfo.py]: I missed that point and will change that.

File: Lib/test/pythoninfo.py (hunk 2 above)
Comment (xdegaye) [reply Lib/test/pythoninfo.py]: Ok.

File: Lib/test/pythoninfo.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/pythoninfo.py]: > I missed that point and will change that.

No problem.

**Comment by vstinner** [comment]
Thanks for the fix Xavier. I backported latest pythoninfo changes (not only this one) to 2.7 and 3.7 branches.

**Pull request merged**